### DISCUSSION TITLE
decimating the mesh in mujoco

### DISCUSSION THREAD

**Discussion by Lhy6900** [google/brax#554] [General]
I used MJX3.1.3 before, recently I have noticed that MJX3.2.4 offer method to obtain contact force. So I change the version.
The same set of code runs smoothly in version 3.1.3, but when running in version 3.2.4, the following error occurs.

```
/home/.conda/envs/jax/lib/python3.9/site-packages/mujoco/mjx/_src/mesh.py:177: UserWarning: Mesh "base" has a coplanar face with more than 20 vertices. This may lead to performance issues and inaccuracies in collision detection. Consider decimating the mesh.
  warnings.warn(
/home/.conda/envs/jax/lib/python3.9/site-packages/mujoco/mjx/_src/mesh.py:177: UserWarning: Mesh "foot" has a coplanar face with more than 20 vertices. This may lead to performance issues and inaccuracies in collision detection. Consider decimating the mesh.
  warnings.warn(
/home/.conda/envs/jax/lib/python3.9/site-packages/mujoco/mjx/_src/mesh.py:52: RuntimeWarning: invalid value encountered in divide
  face_norm = face_norm / np.linalg.norm(face_norm, axis=1).reshape((-1, 1))
```
Is there a method in mjx help decimating the mesh？

**Comment by btaba** [comment]
Hi @lucattyy , you can use `trimesh` or you can also set [`maxhullvert`](https://mujoco.readthedocs.io/en/stable/XMLreference.html#asset-mesh-maxhullvert:~:text=are%20not%20smoothed.-,maxhullvert,-%3A%20int%2C%20%E2%80%9C%2D1) in your MJCF.
